### PR TITLE
Add model validation and specs

### PR DESCRIPTION
**Pull request by preciousuwhubetine**
In this branch, I added validation to the following models:
- `User`
- `Post`
- `Comment` and
- `Like`

I also wrote unit tests using the RSpec testing library for the models listed above.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -3,6 +3,9 @@ class User < ApplicationRecord
   has_many :comments
   has_many :posts, foreign_key: 'author_id'
 
+  validates :name, presence: true
+  validates :posts_counter, numericality: { only_integer: true }
```
Comment (Ismailco) [inline app/models/user.rb]: - [x] Nice work here including only integers, however the requirement state that the integer should be greater than or equal to zero, kindly make sure to implement this so negative numbers are excluded from this validation

File: app/models/post.rb [hunk 2]
```diff
@@ -3,6 +3,10 @@ class Post < ApplicationRecord
   has_many :comments
   has_many :likes
 
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :comments_counter, numericality: { only_integer: true }
+  validates :likes_counter, numericality: { only_integer: true }
```
Comment (Ismailco) [inline app/models/post.rb]: - [x] Same for the `comments_counter` and `likes_counter`, kindly make sure the value of these columns are integers greater than or equal to zero only

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  context '#comments_counter' do
+    it 'comments counter should be greater than or equal to 0' do
+      subject.comments_counter = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'comments counter should accept only integer values' do
+      subject.comments_counter = 2.45
+      expect(subject).to_not be_valid
+    end
```
Comment (Ismailco) [inline spec/models/post_spec.rb]: - [x] [OPTIONAL] You can include tests for `0` and positive integers here so your validations are fully tested

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @preciousbetine, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
**- [ ] While the project is not deployed kindly make sure to remove the `Live Demo` section from your README altogether**

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (preciousuwhubetine) [reply app/models/user.rb]: Ok! Haha... I forgot negative integers existed.

**Review by addod19 — APPROVED** [review]
### Hi @preciousbetine, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Great job making the accurate validations and implementing test to cover them as well :rocket: awesome


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**